packages/features/vault/src/components/vault-side-drawer.tsx

```diff
@@ -30,63 +22,37 @@ export const VaultSideDrawer = ({
   advancedToggle,
   context,
 }: VaultSideDrawerProps) => {
-  const isBlocked = useLeverageBlock();
   const history = useHistory();
-  const { state, updateState } = context;
-  const { vaultAddress, tradeType: _tradeType, canSubmit, confirm } = state;
+  const { state } = context;
+  const { vaultAddress, tradeType: _tradeType } = state;
   const { minDepositRequired } = useVaultProperties(vaultAddress);
   const { minBorrowSize } = useVaultCapacity();
   const tradeType = _tradeType as VaultTradeType;
 
-  const handleCancel = useCallback(() => {
-    history.push(`/vaults/${vaultAddress}`);
-    updateState({ confirm: false });
-  }, [vaultAddress, updateState, history]);
-
-  const handleSubmit = useCallback(() => {
-    updateState({ confirm: true });
-  }, [updateState]);
+  if (!tradeType) return null;
 
-  if (!tradeType) return <ProgressIndicator type="notional" />;
-
-  return confirm ? (
-    <Confirmation2
-      heading={<FormattedMessage {...messages[tradeType].heading} />}
+  return (
+    <TransactionSidebar
       context={context}
-      onCancel={handleCancel}
       showDrawer={false}
-      onReturnToForm={handleCancel}
-    />
-  ) : (
-    <ActionSidebar
       heading={messages[tradeType].heading}
-      helptext={
-        isBlocked
-          ? messages.error.blockedGeoActionHelptext
-          : {
-              ...messages[tradeType].helptext,
-              values: {
-                minDepositRequired,
-                minBorrowSize,
-              },
-            }
-      }
       advancedToggle={advancedToggle}
-      showDrawer={false}
-      canSubmit={canSubmit && !isBlocked}
-      leverageDisabled={isBlocked}
-      cancelRoute={''}
-      CustomActionButton={TradeActionButton}
-      handleSubmit={handleSubmit}
+      onCancelRouteCallback={() => history.push(`/vaults/${vaultAddress}`)}
       hideTextOnMobile={false}
+      helptext={{
+        ...messages[tradeType].helptext,
+        values: {
+          minDepositRequired,
+          minBorrowSize,
+        },
+      }}
     >
       {children}
       {tradeType === 'CreateVaultPosition' ? (
```

Comment:
The result of this ternery expression should be passed in as the `riskComponent`